README.md

```diff
@@ -55,22 +108,51 @@ passed in as parameters) is always nice for users too:
 
 ```yaml
 - hosts: all
-  vars:
-    ssh_foo: "foo foo!"
-    ssh_bar: "progress bar"
-
-  roles:
-    - linux-system-roles.ssh
+  tasks:
+  - name: "Configure ssh clients"
+    include_role:
+      name: linux-system-roles.ssh
+    vars:
+      ssh_user: root
+      ssh:
+        Compression: true
+        GSSAPIAuthentication: no
+        ControlMaster: auto
+        ControlPath: ~/.ssh/.cm%C
+        Match:
+          - Condition: "final all"
+            GSSAPIAuthentication: yes
+        Host:
+          - Condition: example
+            Hostname: example.com
+            User: somebody
+      ssh_ForwardX11: no
 ```
 
 More examples can be provided in the [`examples/`](examples) directory. These
 can be useful especially for documentation.
 
+## Template Generation
+
+The [`ssh_config.j2`](templates/ssh_config.j2) template is programatically
+generated by the scripts in meta. New options should be added to the
+`options_body`.
+
+To regenerate the template, from within the `meta/` directory run:
+`./make_option_list >../templates/ssh_config.j2`
```

Comment:
I would prefer to keep the scripts elsewhere as `meta/` has a specific meaning and use in roles/collections.  Also, anything in the `meta` directory will be copied to the end user release content.  I realize you are copying the structure of this role from ansible-sshd and that's what it does - but I would prefer than everything under linux-system-roles follows the same convention.
That being said - we don't have a convention for a sub-directory for role developer scripts and helper files - @nhosoi @pcahyna any suggestions?  If we use a dot-file directory e.g. `.dev-tools` then we can easily exclude this directory from packaging/conversion.

Reply:
Fine for me to move it into different place or even integrate it better (in make, python setup or whatever would sounds appropriate). I would suggest simple `scripts` or `misc`, but I am fine with anything else.

Reply:
ok - I'll wait to see if Pavel or Noriko have any suggestions.

Reply:
Generally, the problem is that roles are WYSIWYG - what is in the repository gets directly used, without any build or install process. At the same time, the role structure relies on hardcoded directory names, so it is unwise to introduce new, nonstandard directories.
I think that using a leading dot is an acceptable marker for such utility files or directories that are not supposed to be used by the end user. We already have CI configuration files like this and automation in the build process, recently done by @nhosoi , that removes them from the resulting RPMs. While the proposed script has a bit different purpose than the current dotfiles, I think that using this convention is acceptable, as it allows to use the existing automation, and we will probably not come up with a better convention anyway.

Reply:
I have chosen the `.dev-tools` directory for the script and template parts. I updated also reference to them in the `README.md` so they are discoverable in the dot directory.

Reply:
Is it a good idea to document this in the README? Users generally don't care about this. And the dot dir will not get into the RPM or collection, while the README will. I think other roles have a separate document for developer documentation.

Reply:
The network role has a `contributing.md` https://github.com/linux-system-roles/network/blob/main/contributing.md

Reply:
Do you have some example? Is there standardized naming? I understand that README is for both users who want to use this role and developers who want to tweak it, contribute to the role or develop something on top of the current role. But with the RPM packaging, this is getting more complicated.

At this moment, this is something that can be moved as well as the `#Testing` section that I added (which files should probably go also to the `.dev-tools` too).

Reply:
`CONTRIBUTE.md` ?

Reply:
`contributing.md` as in  https://github.com/linux-system-roles/network/blob/main/contributing.md

Reply:
Thanks. Moved now.